chpl/chpl-service/src/main/java/gov/healthit/chpl/upload/listing/normalizer/SvapNormalizer.java

```diff
@@ -23,30 +22,31 @@
 @Component
 @Log4j2
 public class SvapNormalizer {
-    private List<SvapCriteriaMap> svapCriteriaMap = new ArrayList<SvapCriteriaMap>();
+    private SvapDAO svapDao;
 
     @Autowired
     public SvapNormalizer(SvapDAO svapDao) {
-        try {
-            svapCriteriaMap = svapDao.getAllSvapCriteriaMap();
-        } catch (EntityRetrievalException ex) {
-            LOGGER.error("Could not initialize svap criteria map for flexible upload.", ex);
-        }
+        this.svapDao = svapDao;
     }
 
     public void normalize(CertifiedProductSearchDetails listing) {
-        if (!CollectionUtils.isEmpty(listing.getCertificationResults())) {
-            listing.getCertificationResults().stream()
-                .forEach(certResult -> fillInSvapData(certResult));
+        try {
```

Comment:
Instead of putting try/catch around this whole block of code, could you just try/catch the svapDao.getAllSvapCriteriaMap() method? And initialize the svapCriteriaMap object to an empty ArrayList so that if the DAO call fails it's just an empty list. I think that would match the behavior that would have occurred before. Unless you did it this way on purpose for a reason I didn't think of - then I'm happy to learn something!

Reply:
![image](https://user-images.githubusercontent.com/24817202/183940015-f567703d-fc53-4c53-a55c-bd7b85babfb3.png)
`​Local variable svapCriteriaMap defined in an enclosing scope must be final or effectively final​`

I originally wrote the method as you suggested, but this compile error was shown.  I changed it to the current structure to remove the compile error.  I figured since the ​svapDao.getAllSvapCriteriaMap​ is the only method that can throw a ​EntityRetrievalException​ exception, that this structure is effectively the same.